axelor-admin/src/main/java/com/axelor/apps/base/service/app/CreateDataBackupService.java

```diff
@@ -0,0 +1,386 @@
+        }
+        if (tablesNameList.contains(csvInput.getTypeName())) {
+          csvInput.setSearch("self.id = :importId");
+        }
+        if (headerStr != null && headerStr.length() > 0) {
+          headerStr.setLength(headerStr.length() - 1);
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+      LOG.debug("Error From CreateDataBackupService - getCSVData() : " + e.getMessage());
+    }
+    return headerStr.append("\n").append(dataStr);
+  }
+
+  /* Get Header For csv File */
+  private String getMetaModelHeader(Property property, CSVInput csvInput) {
+    String propertyTypeStr = property.getType().toString();
+    String propertyName = property.getName();
+    switch (propertyTypeStr) {
+      case "STRING":
+      case "TEXT":
+      case "BOOLEAN":
+      case "DOUBLE":
+      case "DECIMAL":
+      case "DATE":
+      case "TIME":
+      case "DATETIME":
+      case "BINARY":
+      case "ENUM":
+        return propertyName + SEPARATOR;
+      case "INTEGER":
+      case "LONG":
+        if (propertyName.equalsIgnoreCase("id")) {
+          return "importId" + SEPARATOR;
+        } else {
+          return propertyName + SEPARATOR;
+        }
+      case "ONE_TO_ONE":
+      case "MANY_TO_ONE":
+        String searchId = "importId";
+        if (property.getJavaType().getPackage().getName().equals("com.axelor.meta.db")) {
+          searchId = "id";
+        }
+        csvInput.setSearch("self.importId = :importId");
+        CSVBind csvBind = new CSVBind();
+        String columnName = propertyName + "_importId";
+        csvBind.setColumn(columnName);
+        csvBind.setField(propertyName);
+        csvBind.setSearch("self." + searchId + " = :" + columnName);
+        csvBind.setUpdate(true);
+        csvInput.getBindings().add(csvBind);
+        referenceFlag = true;
+        if (property.isRequired()) {
+          notNullReferenceFlag = true;
+        }
+        return columnName + SEPARATOR;
+      case "ONE_TO_MANY":
+      case "MANY_TO_MANY":
+        String searchId1 = "importId";
+        if (property.getJavaType().getPackage().getName().equals("com.axelor.meta.db")) {
+          searchId1 = "id";
+        }
+        csvInput.setSearch("self.importId = :importId");
+        CSVBind csvBind1 = new CSVBind();
+        String columnName1 = propertyName + "_importId";
+        csvBind1.setColumn(columnName1);
+        csvBind1.setField(propertyName);
+        csvBind1.setSearch("self." + searchId1 + " in :" + columnName1);
+        csvBind1.setUpdate(true);
+        csvBind1.setExpression(columnName1 + ".split('\\\\|') as List");
+        csvInput.getBindings().add(csvBind1);
+        referenceFlag = true;
+        if (property.isRequired()) {
+          notNullReferenceFlag = true;
+        }
+        return columnName1 + SEPARATOR;
+      default:
+        return SEPARATOR;
+    }
+  }
+
+  /* Get Data For csv File */
+  private String getMetaModelData(Property property, Object value) {
```

Comment:
You'd better using an external CSV lib since as soon as a text contains " your CSV is broken at a glance